### ISSUE TITLE
Bug: `SpreadElement` does not allow `NewExpression` parent

### ISSUE THREAD

**Issue by eventualbuddha** [typescript-eslint/typescript-eslint#10023]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

types

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&showAST=es&fileType=.tsx&code=HYUw7gBAggTjCGBPAFAOnQbQLoEog&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import {
  AST_NODE_TYPES,
  ESLintUtils,
  TSESTree,
} from '@typescript-eslint/utils';

export const listener: ESLintUtils.RuleListener = {
  SpreadElement(node: TSESTree.SpreadElement): void {
    switch (node.parent.type) {
      case AST_NODE_TYPES.NewExpression: {
        break;
      }
    }
  },
};
```

### Expected Result

I expect that comparing `node.parent.type` to `AST_NODE_TYPES.NewExpression` to be allowed when `node` is `TSESTree.SpreadElement`, but it isn't because TypeScript errors.

### Actual Result

I get this error from TypeScript:
```
Type 'AST_NODE_TYPES.NewExpression' is not comparable to type 'AST_NODE_TYPES.ArrayExpression | AST_NODE_TYPES.CallExpression | AST_NODE_TYPES.ObjectExpression'. ts(2678)
```

### Additional Info

A couple similar but not identical issues related to this:
- https://github.com/typescript-eslint/typescript-eslint/issues/5421
- https://github.com/typescript-eslint/typescript-eslint/issues/4963

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.6.0` |
| `@typescript-eslint/parser`            | `8.6.0` |
| `@typescript-eslint/rule-tester`       | `8.6.0` |
| `@typescript-eslint/scope-manager`     | `8.6.0` |
| `@typescript-eslint/typescript-estree` | `8.6.0` |
| `@typescript-eslint/type-utils`        | `8.6.0` |
| `@typescript-eslint/utils`             | `8.6.0` |
| `TypeScript`                           | `5.5.3` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `20.16.0` |
